editor/src/components/inspector/inspector.tsx

```diff
@@ -312,8 +313,15 @@ export const Inspector = React.memo<InspectorProps>((props: InspectorProps) => {
   )
 
   const toggleAspectRatioLock = React.useCallback(() => {
-    const actions = selectedViews.map((path) => {
-      return setAspectRatioLock(path, !aspectRatioLocked)
+    const aspectRatioLockPropPath = PP.create(['data-aspect-ratio-locked'])
+    const actions: EditorAction[] = selectedViews.flatMap((path): EditorAction[] => {
+      if (aspectRatioLocked) {
+        return [unsetProperty(path, aspectRatioLockPropPath)]
+      }
+      return [
+        setProperty(path, aspectRatioLockPropPath, jsxAttributeValue(true, emptyComments)),
+        setAspectRatioLock(path, true),
+      ]
```

Comment:
I think this could be simplified significantly by modifying the action handler itself: https://github.com/concrete-utopia/utopia/blob/master/editor/src/components/editor/actions/actions.tsx#L4156 (this would also make the behaviour more consistent if we ever fire that action elsewhere in the editor)

Reply:
done!